### PR TITLE
Move CausalCluster creation to CausalClusterStarter class

### PR DESCRIPTION
**Pull request by eastlondoner**
This is in anticipation of a whole load of changes that I have coming.

This change allows the code for the creation of the Neo4j Causal Cluster to be separate from the CausalCluster class that is created and will mean that the CausalCluster.java file doesn't become too big as both become creation and the created object become more sophisticated.

### CODE REVIEW

File: src/main/java/org/neo4j/junit/jupiter/causal_cluster/CausalClusterStarter.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+
+import org.testcontainers.containers.Neo4jContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.SocatContainer;
+
+public class CausalClusterStarter {
```
Comment (michael-simons) [inline src/main/java/org/neo4j/junit/jupiter/causal_cluster/CausalClusterStarter.java]: Can I have this package private and final please? 

```suggestion
final class CausalClusterStarter {
```

File: src/main/java/org/neo4j/junit/jupiter/causal_cluster/CausalClusterStarter.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+	private static final int DEFAULT_BOLT_PORT = 7687;
+
+	private final Configuration configuration;
+
+	private SocatContainer boltProxy;
+
+	public CausalClusterStarter(Configuration configuration) {
```
Comment (michael-simons) [inline src/main/java/org/neo4j/junit/jupiter/causal_cluster/CausalClusterStarter.java]:
```suggestion
	CausalClusterStarter(Configuration configuration) {
```

**Review by michael-simons — CHANGES_REQUESTED** [review]
Good idea. But I'm a Spring / Java developer by heart. Can we call the thing `CausalClusterFactory`, please?

File: src/main/java/org/neo4j/junit/jupiter/causal_cluster/CausalCluster.java [hunk 3]
```diff
@@ -18,97 +18,38 @@
...
-		try {
-			latch.await();
-		} catch (InterruptedException e) {
-			e.printStackTrace();
-		}
-
-		return new CausalCluster(proxy, cluster);
-	}
-
 	private final SocatContainer proxy;
```
Comment (michael-simons) [inline src/main/java/org/neo4j/junit/jupiter/causal_cluster/CausalCluster.java]: If we introduce the `boltPort`, this should also be named `boltProxy`.

Comment (eastlondoner) [reply src/main/java/org/neo4j/junit/jupiter/causal_cluster/CausalCluster.java]: I actually do that in a subsequent chang, when I introduce a clusterProxy for cluster comms, easy to do it now though :-D

Comment (eastlondoner) [reply src/main/java/org/neo4j/junit/jupiter/causal_cluster/CausalCluster.java]: done

**Comment by eastlondoner** [comment]
@michael-simons I think this should be good to go. I'll squash the commits on merge.

**Review by michael-simons — APPROVED** [review]

**Pull request merged**